### DISCUSSION TITLE
Is it Possible to Interact with MN Chain via an API

### DISCUSSION THREAD

**Discussion by aj-agoric** [Agoric/agoric-sdk#8946] [Q&A]
editing since I can not close this. I will re-post as needed

**Comment by dckc** [comment]
There is a REST / Swagger / OpenAPI API; see:

 - [Integrating with Agoric Network \| Documentation](https://docs.agoric.com/guides/integration/chain-integration.html)

That REST API is based on a gRPC / Protobuf API; for example, see `MsgProvision` in [agoric/swingset/msgs.proto](https://github.com/Agoric/agoric-sdk/blob/master/golang/cosmos/proto/agoric/swingset/msgs.proto)

[cosmjs](https://github.com/cosmos/cosmjs) is designed to facilitate use of the REST and gRPC protocols / APIs. That's what we use in our production dapps.

I've been having some fun with batch / CLI usage; see:

 - https://github.com/Agoric/agoric-sdk/discussions/8551

**Comment by aj-agoric** [comment]
Thanks @dckc helpful start. I wanted to re-energize my question here to keep track of the full ask.

App explanation: an agoric type sportsbook thats non profit and allows users to setup a local instance that benefits a given charity. Same options and features of a DraftKings or FanDuel except, when you lose, its benefiting your charity of choice and not just billion dollar casinos. (when you win you are paid the same).

Essentially I have this [working app](https://gorabook.com/) here that I would like to bring on chain in a phased roll out approach.

Phase 1: I simply would like to host the wallet on chain. So the odds calculations, determining winners, and other calculations remain off chain in Web2.. however the wallets and funds are on chain

[example flow is more mapped here](https://miro.com/app/board/uXjVNr08MUI=/). 
essentially - 
new customer signs up >> we send a message to the chain via api saying please create/provision this wallet >> chain responds with some unique wallet ID
Customer places a 5bld wager >> we ask the chain to transfer 5bld from users's wallet to Admin Pot Wallet

Customer wins wager >> we ask the chain to tranfer 8 bld from admin pot to user's wallet

Charity payout >> we ask the chain to transfer XXbld to from admin wallet to charity owned wallet.

App is working now using "credits" a made up currency lol but is essentially dollars. 

Future versions I have mapped can have the enitre operation on chain, however thats further down the road. see the [phased roll out plan](https://docs.google.com/spreadsheets/d/11_g78Ne6Shrrv2fBcQ_h0Z67iZiRoCgRoBDiNabcGrI/edit#gid=0) for my best approach

In the ~2cnd implementation it would be really cool to use more of our orchestration.. The local wallet should be able to hold multiple currency that can be used freely to play. Example bet 2bld on one game, 3 IST on another, 1 TIA on a third.  

again the first and most helpful version is creating wallets and orchestrating transfers from my web2 app.
